### PR TITLE
refactor(*): lint and refactor

### PR DESCRIPTION
**Pull request by Gheoan**

### CODE REVIEW

File: src/array-change-records.js [hunk 1]
```diff
@@ -396,8 +397,7 @@ function createInitialSplices(array, changeRecords) {
       mergeSplice(splices, index, [record.oldValue], record.type === 'delete' ? 0 : 1);
       break;
     default:
-      console.error('Unexpected record type: ' + JSON.stringify(record)); // eslint-disable-line
-      break;
+      throw new Error(`Unexpected record type: ${JSON.stringify(record)}`);
```
Comment (atsu85) [inline src/array-change-records.js]: This doesn't look like just refactoring at first glance ;)

Comment (Gheoan) [inline src/array-change-records.js]: This is linting.

**Comment by EisenbergEffect** [comment]
@jdanyow Please review.

**Comment by jdanyow** [comment]
@Gheoan could you please write a test for this and change the commit message to something that indicates what changed.  `fix(array-change-records): throw on unexpected record type` perhaps?

**Comment by Gheoan** [comment]
I undid the change to `array-change-records` and added one test for `AccessMember`. The rest of the changes seem to be well covered by tests.

**Pull request merged**